### ISSUE TITLE
Aff compatability

### ISSUE THREAD

**Issue by athanclark** [Thimoteus/purescript-promises#8]
Would you be interested in making a `Promise <~> Aff` isomorphic natural transformation in this library? It's really simple to do:

```
toAff :: forall a. Promise a -> Aff a
toAff p = makeAff \resolve -> nonCanceler <$ runPromise (resolve <<< Right) (resolve <<< Left) p

fromAff :: forall a. Aff a -> Promise a
fromAff x = promise \ok bad -> runAff_ (either bad ok) x
```

The only non-isomorphic artifact is cancelation in the Aff context - but that's an additional feature to general purpose asynchronous execution; so I think just a note in the documentation for these functions would be satisfactory for end users.

**Comment by rockymadden** [comment]
@athanclark I love this idea and something I was looking for myself. Either in the lib proper or simply as documentation.

**Comment by athanclark** [comment]
@rockymadden I think it's omitted to avoid the transitive dependencies

**Comment by Thimoteus** [comment]
There's another PS library for going between JS promises and Aff: https://github.com/nwolverson/purescript-aff-promise

This is more of a way to use promises using `do` notation (without depending on the Aff runtime), or for people new to PS to use promises using somewhat familiarish notation.

**Comment by rockymadden** [comment]
Makes sense, thank you. I was and am using said library. What I initially didn't understand was how `Deferred` cannot be discharged except with a couple functions that also force evaluation (which makes total sense upon reflection). So, when I started using `fromAff` from above, I had to put a type class constraint on it -- and all functions that used it, and up, and up the chain. I ended up going back to purescript-aff-promise and doing some eww in JS for my use case.
